packages/backend/src/open_payments/payment/incoming_remote/service.ts

```diff
@@ -0,0 +1,150 @@
+import {
+  AuthenticatedClient,
+  IncomingPayment as OpenPaymentsIncomingPayment,
+  isNonInteractiveGrant,
+  AccessAction,
+  PaymentPointer as OpenPaymentsPaymentPointer
+} from 'open-payments'
+import { Grant } from '../../grant/model'
+import { GrantService } from '../../grant/service'
+import { BaseService } from '../../../shared/baseService'
+import { Amount, serializeAmount } from '../../amount'
+
+interface CreateRemoteIncomingPaymentArgs {
+  paymentPointerUrl: string
+  description?: string
+  expiresAt?: Date
+  incomingAmount?: Amount
+  externalRef?: string
+}
+
+export interface RemoteIncomingPaymentService {
+  create(
+    args: CreateRemoteIncomingPaymentArgs
+  ): Promise<OpenPaymentsIncomingPayment>
```

Comment:
```suggestion
  ): Promise<OpenPaymentsIncomingPayment | IncomingPaymentError>
```
to be able to return `IncomingPaymentError.UnknownPaymentPointer` (instead of throwing) and return the corresponding code and message in the `createReceiver` resolver
https://github.com/interledger/rafiki/blob/752efd71a7ee4cfa3200a9757db7a09242ba7606/packages/backend/src/open_payments/payment/incoming/errors.ts#L1-L2

Reply:
Good idea

Reply:
Actually, do you mean
```ts
// ReceiverService
async function createLocalIncomingPayment(
  deps: ServiceDependencies,
  args: CreateReceiverArgs,
  paymentPointer: PaymentPointer
): Promise<OpenPaymentsIncomingPayment | IncomingPaymentError>
```
instead? `RemotePaymentService` can only return the `OpenPaymentsIncomingPayment` since it uses the client

Reply:
I wonder if having a custom Error object with something like a `userMessage` property would be a better approach

Reply:
No, I think it needs to be in `RemotePaymentService`.
The payment pointer look up in `getGrant` should somehow bubble up an error to allow the resolver to return `4XX`.

Reply:
> I wonder if having a custom Error object with something like a `userMessage` property would be a better approach

That'd probably work too. I'm just thinking through the lens of the other services' `*Error`s.

Reply:
Makes sense, I'll give it a try :+1: